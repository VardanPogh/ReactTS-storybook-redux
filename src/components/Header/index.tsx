import React from 'react'
import clsx from 'clsx'
import { makeStyles, createStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import Container from '@material-ui/core/Container'
import { useHistory } from 'react-router-dom'

import Button from '../Button'

const useStyles = makeStyles(() =>
  createStyles({
    container: {
      marginTop: 20,
      marginBottom: 70,
      '@media (max-width: 375px)': {
        marginBottom: 0,
      },
    },
    logo: {
      border: 0,
      backgroundColor: 'transparent',
      cursor: 'pointer',
      '& img': {
        width: 200,
      },
    },
    buttonSupport: {
      marginRight: 5,
    },
    buttonRegister: {
      marginRight: 17,
    },
    linkButtons: {
      '@media (max-width: 768px)': {
        marginBottom: 10,
      },
    },
  })
)

interface Props {
  haveButtons?: boolean
  isLoggedIn?: boolean
}

const Header: React.FC<Props> = ({ haveButtons, isLoggedIn }) => {
  const classes = useStyles()
  const history = useHistory()
  return (
    <Container className={classes.container} maxWidth="lg">
      <Grid container spacing={3}>
        <Grid item xs sm={4} md={6} lg={7}>
          <button
            className={classes.logo}
            type="button"
            onClick={(): void => history.push('/')}
          >
            <img src="img/logo.png" alt="logo" />
          </button>
        </Grid>
        {haveButtons && (
          <Grid item xs sm={8} md={6} lg={5}>
            <Button
              className={clsx(classes.linkButtons)}
              onClick={(): void => history.push('/support')}
              text="Get a Demo"
            />
            <Button
              onClick={(): void => history.push('/support')}
              className={clsx(classes.buttonSupport, classes.linkButtons)}
              text="Get support"
            />
            <Button
              onClick={(): void => history.push('/register')}
              className={classes.buttonRegister}
              text="Register"
              variant="outlined"
            />
            <Button
              onClick={(): void => history.push('/login')}
              text="Login"
              variant="contained"
            />
          </Grid>
        )}
        {isLoggedIn && (
          <Grid item xs={12} sm={2} md={2} lg={2}>
            <p>Wendel Costa</p>
          </Grid>
        )}
      </Grid>
    </Container>
  )
}

export default Header
