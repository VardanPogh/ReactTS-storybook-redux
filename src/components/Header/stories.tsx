import React from 'react'
import { withKnobs, boolean } from '@storybook/addon-knobs'
import Header from '.'

export default {
  title: 'Header',
  component: Header,
  decorators: [withKnobs],
}

export const Default: React.FC = () => (
  <Header haveButtons={boolean('Have Buttons', true)} />
)
