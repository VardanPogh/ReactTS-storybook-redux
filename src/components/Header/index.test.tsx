import React from 'react'
import { render } from '@testing-library/react'

import Header from '.'

describe('Header Test', () => {
  test('Render Header with buttons navigation', () => {
    const { container } = render(<Header haveButtons />)
    expect(container).toMatchSnapshot()
  })
  test('Render Header without buttons navigation', () => {
    const { container } = render(<Header haveButtons={false} />)
    expect(container).toMatchSnapshot()
  })
})
