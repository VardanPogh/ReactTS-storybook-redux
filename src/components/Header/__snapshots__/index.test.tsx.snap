// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Render Header with buttons navigation 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-7"
      >
        <button
          class="makeStyles-logo-2"
          type="button"
        >
          <img
            alt="logo"
            src="img/logo.png"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true MuiGrid-grid-sm-8 MuiGrid-grid-md-6 MuiGrid-grid-lg-5"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-asLink-7 makeStyles-linkButtons-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Get a Demo
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-asLink-7 makeStyles-buttonSupport-3 makeStyles-linkButtons-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Get support
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-outlined-8 makeStyles-buttonRegister-4"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Register
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-contained-6"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header Test Render Header without buttons navigation 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-container-9 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-7"
      >
        <button
          class="makeStyles-logo-10"
          type="button"
        >
          <img
            alt="logo"
            src="img/logo.png"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
